core/workspace_svg.ts

```diff
@@ -2541,6 +2541,69 @@ export class WorkspaceSvg extends Workspace implements IASTNodeLocationSvg {
       this.removeClass('blocklyReadOnly');
     }
   }
+
+  /**
+   * Scrolls the provided bounds into view.
+   *
+   * In the case of small workspaces/large bounds, this function prioritizes
+   * getting the top left corner of the bounds into view. It also adds some
+   * padding around the bounds to allow the element to be comfortably in view.
+   *
+   * @internal
+   * @param bounds A rectangle to scroll into view, as best as possible.
+   */
+  scrollBoundsIntoView(bounds: Rect) {
+    if (Gesture.inProgress()) {
+      // This can cause jumps during a drag and it only suited for keyboard nav.
+      return;
+    }
+    const scale = this.getScale();
+
+    const rawViewport = this.getMetricsManager().getViewMetrics(true);
+    const viewport = new Rect(
+      rawViewport.top,
+      rawViewport.top + rawViewport.height,
+      rawViewport.left,
+      rawViewport.left + rawViewport.width,
+    );
+
+    if (
+      bounds.left >= viewport.left &&
+      bounds.top >= viewport.top &&
+      bounds.right <= viewport.right &&
+      bounds.bottom <= viewport.bottom
+    ) {
+      // Do nothing if the block is fully inside the viewport.
+      return;
+    }
+
+    // Add some padding to the bounds so the element is scrolled comfortably
+    // into view.
+    bounds = bounds.clone();
+    bounds.top -= 10;
```

Comment:
nit: magic number. Consider making the padding a param or at least a constant somewhere.